src/integrationtests/java/com/aws/greengrass/integrationtests/ipc/SubscribeToCertificateUpdatesTest.java

```diff
@@ -0,0 +1,212 @@
+                new SubscribeToCertificateUpdatesRequest();
+        subscribeToCertificateUpdatesRequest.setCertificateOptions(new CertificateOptions().withCertificateType(
+                CertificateType.SERVER));
+        return subscribeToCertificateUpdatesRequest;
+    }
+
+    @BeforeEach
+    void setup() {
+        kernel = new Kernel();
+        kernel.getContext().put(GreengrassServiceClientFactory.class, clientFactory);
+        lenient().when(clientFactory.getGreengrassV2DataClient()).thenReturn(client);
+    }
+
+    private void startNucleusWithConfig(String configFileName) throws InterruptedException {
+        startNucleusWithConfig(configFileName, State.RUNNING);
+    }
+
+    private void startNucleusWithConfig(String configFileName, State expectedServiceState) throws InterruptedException {
+        CountDownLatch authServiceRunning = new CountDownLatch(1);
+        kernel.parseArgs("-r", rootDir.toAbsolutePath().toString(), "-i",
+                getClass().getResource(configFileName).toString());
+        listener = (GreengrassService service, State was, State newState) -> {
+            if (service.getName().equals(ClientDevicesAuthService.CLIENT_DEVICES_AUTH_SERVICE_NAME) &&
+                    service.getState().equals(expectedServiceState)) {
+                authServiceRunning.countDown();
+            }
+        };
+        kernel.getContext().addGlobalStateChangeListener(listener);
+        kernel.launch();
+        assertThat(authServiceRunning.await(30L, TimeUnit.SECONDS), is(true));
+        kernel.getContext().removeGlobalStateChangeListener(listener);
+    }
+
+    @AfterEach
+    void afterEach() {
+        LogConfig.getRootLogConfig().reset();
+        kernel.shutdown();
+    }
+
+    @Test
+    void GIVEN_two_brokers_WHEN_subscribed_to_certificate_updates_THEN_both_receive_updates_on_stream()
+            throws Exception {
+        startNucleusWithConfig("cda.yaml");
+        Map<String, String> broker1Certs = new HashMap<>();
+        Map<String, String> broker2Certs = new HashMap<>();
+        try (EventStreamRPCConnection connection = IPCTestUtils.getEventStreamRpcConnection(kernel,
+                "BrokerSubscribingToCertUpdates")) {
+            GreengrassCoreIPCClient ipcClient = new GreengrassCoreIPCClient(connection);
+            SubscribeToCertificateUpdatesRequest request = getSampleSubsRequest();
+            Pair<CompletableFuture<Void>, Consumer<CertificateUpdate>> cb = asyncAssertOnConsumer((m) -> {
+                broker1Certs.put("certificate", m.getCertificate());
+                broker1Certs.put("ca-certificate", m.getCaCertificates().get(0));
+                assertThat(m.getCertificate(), containsString("-----BEGIN CERTIFICATE-----"));
+                assertThat(m.getCaCertificates().get(0), containsString("-----BEGIN CERTIFICATE-----"));
+                assertThat(m.getPrivateKey(), containsString("-----BEGIN PRIVATE KEY-----"));
```

Comment:
contains is good for sure, but I think you can make this even more specific with a startsWith and endsWith